provider/database_db_homes_resource.go

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) 2017, Oracle and/or its affiliates. All rights reserved.
+
+package provider
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/oracle/bmcs-go-sdk"
+
+	"github.com/oracle/terraform-provider-oci/crud"
+)
+
+func DBHomeResource() *schema.Resource {
+	return &schema.Resource{
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: crud.DefaultTimeout,
+		Create:   createDBHome,
+		Read:     readDBHome,
+		Update:   updateDBHome,
+		Delete:   deleteDBHome,
+		Schema: map[string]*schema.Schema{
+			//Required
+			"database": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
```

Comment:
This should probably just share the same schema as database_database_data_source.go (as should the database schema in database_db_system*.go). We should also be reading in service data about the database in SetData... and we might not want to use the full schema that includes purely computed fields until we do that.